client/tmclient/ui.py

```diff
@@ -178,11 +179,61 @@ def unfocus(self):
         self.tab_header = TabHeader(self.tab_header.label, self.tab_header.position, False)
 
 
+class ColorText(urwid.Text):
+    """
+    Turns a string into a text widget with colorized sections.
+    Strings use a format of "foo {color name}bar{/}"
+    """
+
+    def __init__(self, s, align='left', wrap='space', layout=None):
+        parts = re.findall('(\{[^\}]+\})|([^\{]*)', s)
```

Comment:
For REs that are going to be evaluated a lot, i tend to prefer compiling them once as a module global:

```python
COLOR_PARTS_RE = re.compile('(\{[^\}]+\})|([^\{]*)')
```

and then using `COLOR_PARTS_RE.findall(s)`